### PR TITLE
zsh: set autosuggestion color

### PR DESCRIPTION
**Pull request by teto**
### Description

syntax for ref https://askubuntu.com/questions/1393204/changing-zsh-autosuggestions-color
haven't played much with it

### Checklist

- [ ] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ ] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -220,6 +220,21 @@ let
...
+  autosuggestionModule = types.submodule {
+
+    options = {
+        enable = mkOption {
+          default = false;
+          description = "Enable zsh autosuggestions";
+        };
+        highlight = mkOption {
+          default = "fg=cyan";
+          description = "Enable zsh autosuggestions";
```
Comment (rycee) [inline modules/programs/zsh.nix:232]: I imagine that the description here needs to be updated to match the option name.

File: modules/programs/zsh.nix [hunk 2]
```diff
@@ -368,6 +379,13 @@ in
         description = "Options related to zsh-history-substring-search.";
       };
 
+      autosuggestion = mkOption {
```
Comment (rycee) [inline modules/programs/zsh.nix:382]: I think you can skip the submodule. in this case it should be possible to simply do something like

```
autosuggestion = {
  enable = mkOption …;
  highlight = mkOption …;
};

**Comment by rycee** [comment]
Thanks! Added some minor comments 🙂k

File: modules/programs/zsh.nix (hunk 2 above)
Comment (teto) [reply modules/programs/zsh.nix:382]: ha that's how I had first implemented it than I saw the highlighting module and decided to copy whatthey did. I will revert that.

Comment (rycee) [reply modules/programs/zsh.nix:382]: Yeah, we should probably also simplify the highlighting options in the same way. The submodule looks unnecessary.

**Comment by teto** [comment]
I think it's good but as I went back and forth between the different implementations, I would prefer for someone else to make sure it works and merge.

**Comment by rycee** [comment]
@teto I made some minor changes, like adding `type` fields to the options and making the highlight optional (using `nullOr`). Please have a look and if you think it is OK then I think we can merge it.

**Comment by rycee** [comment]
Merged now, hopefully that's OK 🙂

**Comment by mfroeh** [comment]
Maybe I'm overlooking something, but for me `programs.zsh.autosuggestion.highlight` has no effect.
It sets `AUTOSUGGESTION_HIGHLIGHT_COLOR` instead of `ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE` as required by 
[zsh-autosuggestions](https://github.com/zsh-users/zsh-autosuggestions?tab=readme-ov-file#configuration).

**Pull request merged**